pilot/pkg/serviceregistry/kube/pod.go

```diff
@@ -107,7 +107,7 @@ func (pc *PodCache) event(obj interface{}, ev model.Event) error {
 			// delete only if this pod was in the cache
 			if pc.keys[ip] == key {
 				delete(pc.keys, ip)
-				if pc.c != nil {
+				if pc.c.XDSUpdater != nil {
```

Comment:
Can you double check this? `pc.c` is a reference field and can be nil.
Maybe all the if conditions above (`if pc.c.XDSUpdater != nil`) are the incorrect ones.